static/js/message_view_header.js

```diff
@@ -105,6 +105,8 @@ function bind_title_area_handlers() {
             e.preventDefault();
             e.stopPropagation();
         }
+        // reset text
+        $("#search_query").val("");
```

Comment:
I don't think this placement is correct; won't it break the default value of the search query for non-recent-topics views?  E.g. if you start search within the stream, we want the existing stream value to be there.

Reply:
Yes @timabbott , I figured that out with the help of @YashRE42 . I had got confused, but I'm clear now, will push the changes soon.